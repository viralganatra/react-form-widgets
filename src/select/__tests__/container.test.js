import React from 'react';
import SelectContainer from '../container';

jest.mock('react-clickoutside');
jest.mock('../label-widget', () => jest.fn());
jest.mock('../group-items-container', () => jest.fn());
jest.mock('../group-items-widget', () => jest.fn());

const props = {
    items: [
        {
            label: 'Test 1',
            value: 1,
        },
        {
            label: 'Test 2',
            value: 2,
        },
    ],
};

describe('Container', () => {
    it('should render correctly', () => {
        const wrapper = shallow(<SelectContainer {...props} />);
        const wrapperSelected = shallow(
            <SelectContainer {...props} {...props.items[0]} />,
        );
        const wrapperEnableSelectionReset = shallow(
            <SelectContainer {...props} enableSelectionReset={true} />,
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapperSelected).toMatchSnapshot();
        expect(wrapperEnableSelectionReset).toMatchSnapshot();

        wrapper.find('div').childAt(0).simulate('click');

        expect(wrapper).toMatchSnapshot();

        wrapper.find('div').childAt(1).simulate('clickOutside');

        expect(wrapper).toMatchSnapshot();
    });

    it('should validate the prop types', () => {
        const wrapperInvalidLabel = () => shallow(
            <SelectContainer {...props} label={2} />,
        );

        const wrapperInvalidValue = () => shallow(
            <SelectContainer {...props} value={[]} />,
        );

        const wrapperInvalidItems = () => shallow(
            <SelectContainer items={[{ label: 2, value: [] }]} />,
        );

        expect(wrapperInvalidLabel).toThrowErrorMatchingSnapshot();
        expect(wrapperInvalidValue).toThrowErrorMatchingSnapshot();
        expect(wrapperInvalidItems).toThrowErrorMatchingSnapshot();
    });

    it('should set the selected item in state when onItemSelection is called', () => {
        const wrapper = shallow(<SelectContainer {...props} />);

        wrapper.find('div').childAt(1).simulate('itemSelection', props.items[1]);

        expect(wrapper).toMatchSnapshot();
    });

    it('should call the onChange event from props when onItemSelection is called', () => {
        const methodSpy = jest.fn();
        const wrapper = shallow(<SelectContainer {...props} onChange={methodSpy} />);

        wrapper.find('div').childAt(1).simulate('itemSelection', props.items[1]);

        expect(methodSpy).toHaveBeenCalledWith(props.items[1]);
    });
});
