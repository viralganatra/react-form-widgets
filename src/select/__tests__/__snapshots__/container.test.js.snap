exports[`Container should render correctly 1`] = `
<div>
  <mockConstructor
    onClick={[Function]} />
  <Mock
    items={
      Array [
        Object {
          "label": "Test 1",
          "value": 1,
        },
        Object {
          "label": "Test 2",
          "value": 2,
        },
      ]
    }
    onClickOutside={[Function]}
    onItemSelection={[Function]}
    shouldDisplay={false} />
</div>
`;

exports[`Container should render correctly 2`] = `
<div>
  <mockConstructor
    label="Test 1"
    onClick={[Function]} />
  <Mock
    items={
      Array [
        Object {
          "label": "Test 1",
          "value": 1,
        },
        Object {
          "label": "Test 2",
          "value": 2,
        },
      ]
    }
    onClickOutside={[Function]}
    onItemSelection={[Function]}
    shouldDisplay={false} />
</div>
`;

exports[`Container should render correctly 3`] = `
<div>
  <mockConstructor
    onClick={[Function]}
    onResetSelection={[Function]} />
  <Mock
    items={
      Array [
        Object {
          "label": "Test 1",
          "value": 1,
        },
        Object {
          "label": "Test 2",
          "value": 2,
        },
      ]
    }
    onClickOutside={[Function]}
    onItemSelection={[Function]}
    shouldDisplay={false} />
</div>
`;

exports[`Container should render correctly 4`] = `
<div>
  <mockConstructor
    onClick={[Function]} />
  <Mock
    items={
      Array [
        Object {
          "label": "Test 1",
          "value": 1,
        },
        Object {
          "label": "Test 2",
          "value": 2,
        },
      ]
    }
    onClickOutside={[Function]}
    onItemSelection={[Function]}
    shouldDisplay={true} />
</div>
`;

exports[`Container should render correctly 5`] = `
<div>
  <mockConstructor
    onClick={[Function]} />
  <Mock
    items={
      Array [
        Object {
          "label": "Test 1",
          "value": 1,
        },
        Object {
          "label": "Test 2",
          "value": 2,
        },
      ]
    }
    onClickOutside={[Function]}
    onItemSelection={[Function]}
    shouldDisplay={false} />
</div>
`;

exports[`Container should set the selected item in state when onItemSelection is called 1`] = `
<div>
  <mockConstructor
    label="Test 2"
    onClick={[Function]} />
  <Mock
    items={
      Array [
        Object {
          "label": "Test 1",
          "value": 1,
        },
        Object {
          "label": "Test 2",
          "value": 2,
        },
      ]
    }
    onClickOutside={[Function]}
    onItemSelection={[Function]}
    shouldDisplay={false} />
</div>
`;

exports[`Container should validate the prop types 1`] = `
"Warning: Failed prop type: Invalid prop \`label\` of type \`number\` supplied to \`SelectWidget\`, expected \`string\`.
    in SelectWidget"
`;

exports[`Container should validate the prop types 2`] = `
"Warning: Failed prop type: Invalid prop \`value\` supplied to \`SelectWidget\`.
    in SelectWidget"
`;

exports[`Container should validate the prop types 3`] = `
"Warning: Failed prop type: Invalid prop \`items[0].value\` supplied to \`SelectWidget\`.
    in SelectWidget"
`;
